src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -418,7 +419,42 @@ func (c controller) postInstallConfigs(ctx context.Context) error {
 		if err = c.updatePendingOLMOperators(); err != nil {
 			return errors.Errorf("Timeout while waiting for some of the operators and not able to update its state")
 		}
-		c.log.Warnf("Timeout while waiting for OLM operators be installed")
+		c.log.WithError(err).Warnf("Timeout while waiting for OLM operators be installed")
+	}
+
+	// Aplly post install manifests:
+	err = c.applyPostInstallManifests()
```

Comment:
Users do not see logs .
If we fail applying manifests for an operator, we should probably mark that operator as failed, not the entire cluster installation

Reply:
Sure, we do that, that's why we don't return error at line 428. We update the status of the operator [here](https://github.com/openshift/assisted-installer/blob/4f5ad6a52b0fa1bcacb19be70aaeedb09870dda0/src/assisted_installer_controller/assisted_installer_controller.go#L739)

Reply:
The reason we don't return error here and also on line 422 is because the [PostInstallConfig](https://github.com/openshift/assisted-installer/blob/bc513cf9e7188afac3fbbc4d55e8228542347ee4/src/assisted_installer_controller/assisted_installer_controller.go#L851) method decide whether to set the installation as successful based on the error returned from this method. So we do not return the error on purpose. If we would return error here the installation would be marked as failed. If we just log the error here and we update the status of the operators in assisted-service API with related error, we can mark the installation as successful, and report what's wrong with monitored operators.

Reply:
@machacekondra right now controller is not deciding if installation is successful, it only decides if it is not.
So if you will return error here it will not really matter. In any case we have ticket to remove send complete api from here.
@YuviGold can you please assist here?

Reply:
But main problem that this is not the right place to add this code. It must be before 
https://github.com/openshift/assisted-installer/pull/271/files#diff-1da579fda3bc689940661d48dd744a46f23824270427207cfae4f8c7448eddd2L414
or even part of waitForOLMOperators (run only once on success though)

Reply:
Done

Reply:
Based on the discussion I understand, this is really not important. So should I change or keep? @tsorya